WalletWasabi/Services/UpdateManager.cs

```diff
@@ -0,0 +1,238 @@
+using Newtonsoft.Json.Linq;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using WalletWasabi.Logging;
+using WalletWasabi.Microservices;
+using WalletWasabi.Models;
+using WalletWasabi.Tor.Http;
+
+namespace WalletWasabi.Services;
+
+public class UpdateManager
+{
+	private string InstallerName { get; set; } = "";
+	private const byte MaxTries = 3;
+	private const string ReleaseURL = "https://api.github.com/repos/zkSNACKs/WalletWasabi/releases/latest";
+
+	public UpdateManager(string dataDir, bool downloadNewVersion, IHttpClient httpClient)
+	{
+		DownloadsDir = Path.Combine(dataDir, "Downloads");
+		HttpClient = httpClient;
+		DownloadNewVersion = downloadNewVersion;
+	}
+
+	public async void UpdateStatusChangedAsync(UpdateStatus updateStatus)
+	{
+		var tries = 0;
+		bool updateAvailable = !updateStatus.ClientUpToDate || !updateStatus.BackendCompatible;
+		Version targetVersion = updateStatus.ClientVersion;
+		if (!updateAvailable)
+		{
+			return;
+		}
+		if (DownloadNewVersion)
+		{
+			Logger.LogInfo($"Trying to download new version: {targetVersion}");
+			do
+			{
+				tries++;
+				try
+				{
+					bool isReadyToInstall = await GetInstallerAsync(targetVersion).ConfigureAwait(false);
+					Logger.LogInfo($"Version {targetVersion} downloaded successfuly.");
+					updateStatus.IsReadyToInstall = isReadyToInstall;
+					break;
+				}
+				catch (Exception ex)
+				{
+					Logger.LogError($"Geting version {targetVersion} failed with error.", ex);
+				}
+			} while (tries < MaxTries);
+		}
+
+		UpdateAvailableToGet?.Invoke(this, updateStatus);
+	}
+
+	private async Task<bool> GetInstallerAsync(Version targetVersion)
+	{
+		if (CheckIfInstallerDownloaded())
```

Comment:
This method should get the version as well as it needs to verify if the newest was downloaded. Something is missing here.